app/models/user.rb

```diff
@@ -1,5 +1,6 @@
 class User < ActiveRecord::Base
   authenticates_with_sorcery!
+  before_create :create_organiser_token
```

Comment:
the check cant be possible as, new user is created with a token, so we save the new user as organiser in update fn after filling the form.